dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/injection/EEValidationUtils.java

```diff
@@ -209,35 +209,31 @@ private static void validateJndiLookup(InitialContext c, String lookupString, An
         while (contents.hasMore()) {
             NameClassPair pair = contents.next();
             if (name.equals(pair.getName())) {
-                try {
-                    String className = pair.getClassName();
-                    Class<?> jndiClass = cdiArchive.getClassLoader().loadClass(className);
-                    if ("javax.resource.cci.ConnectionFactory".equals(className)) {
-                        try {
-                            Object o = c.lookup(lookupName);
-                            if (o != null) {
-                                jndiClass = o.getClass();
-                            }
-                        } catch (RuntimeException e) {
-                            // An error occurred while getting the object from JNDI. This may happen
-                            // at this early point in the initialisation process, but if so we just
-                            // skip validation.
-                        }
-                    } else {
-                        try {
-                            jndiClass = cdiArchive.getClassLoader().loadClass(className);
-                        } catch (ClassNotFoundException ex) {
-                            // Couldn't load the jndiClass name, can't validate
+                String className = pair.getClassName();
+                Class<?> jndiClass = null;
+                if ("javax.resource.cci.ConnectionFactory".equals(className)) {
+                    try {
+                        Object o = c.lookup(lookupName);
+                        if (o != null) {
+                            jndiClass = o.getClass();
                         }
+                    } catch (RuntimeException e) {
```

Comment:
I really don't remember writing this code but catching a RuntimeException here seems a bit stupid?!